IoTHubManager/DevicePropertiesTest.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace IoTHubManager
+{
+    [Collection("IoTHub Manager Tests")]
+    public class DevicePropertiesTest : IDisposable
+    {
+        private HttpRequestWrapper Request;
+        private HttpRequestWrapper PropertyRequest;
+        public string deviceId = "";
+        public string eTag = "";
+        public JObject device;
+
+        public DevicePropertiesTest()
+        {
+            this.Request = new HttpRequestWrapper(Constants.IOT_HUB_ADDRESS, Constants.Urls.DEVICE_PATH);
+            this.PropertyRequest = new HttpRequestWrapper(Constants.IOT_HUB_ADDRESS, Constants.Urls.DEVICE_PROPERTIES_PATH);
+            this.CreateDevice();
+        }
+
+        private void CreateDevice()
+        {
+            var DEVICE_TEMPLATE = System.IO.File.ReadAllText(Constants.Path.DEVICE_FILE_AUTO_GEN_AUTH);
+            var device = DEVICE_TEMPLATE.Replace(Constants.Keys.DEVICE_ID, "");
+
+            var response = this.Request.Post(device);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            this.device = JObject.Parse(response.Content);
+            this.deviceId = this.device["Id"].ToString();
+        }
+
+        /**
+        * Creates Job for tagging on devices and 
+        * checks the job status using polling 
+        * mechanism to verify job completion.
+        */
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void DevicePropertiesListUpdated_IfDeviceUpdated()
+        {
+            //Arrange
+            string jobId = Guid.NewGuid().ToString();
+            string devicePropertyKey = Guid.NewGuid().ToString();
+            string devicePropertyValue = Guid.NewGuid().ToString();
+            this.device["Tags"][devicePropertyKey] = devicePropertyValue;
+
+            // Act
+            var response = Request.Put(this.deviceId, this.device);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Thread.Sleep(Constants.Jobs.WAIT*5);
+
+            // Asserts
+            AssertKeyExistsInPropertyList(devicePropertyKey);
+            // DeleteDeviceProperty(devicePropertyKey); Deleting tags NOT working
+            // Thread.Sleep(Constants.Jobs.WAIT * 5);
+            // AssertKeyDoesNotExistsInPropertyList(devicePropertyKey);
+            // Deleting tags si currently not updating the device property list.
```

Comment:
typo: is instead of si.
`There is a bug in delete tags where it appends new tags but doesn't remove deleted tags. There is a bug open for it`